Team-Hack-IT/OmniAuth
# Migrate supabase

## Summary
This pull request adds a sleuth of improvement and features to the backend service.

## Changes made
This pull request makes the following changes: -
* Integrated supabase for database and storage
* Integrated Tron to interact with the smart contract.
* Implemented logging and rate limiting.
* Improved the unit tests to verify the image comparison works well.
* Refactored the controllers and utility functions.
* Improved code quality and enforced style and quality checks before commits.
* Improved the API docs and README.md.
* Improved error handling.
* General Bug fixes.

## How to test
To test these changes:
1. Checkout this branch.
2. Run `npm install` to install any new dependencies.
3. Run `npm test` to run the new unit tests. 4. Manually test the payment processing system to ensure that transactions are recorded correctly.
